tests/lib/nested.sh

```diff
@@ -117,6 +129,10 @@ start_nested_core_vm_unit(){
     if [ "${ENABLE_TPM}" = "true" ]; then
         if ! snap list swtpm-mvo > /dev/null; then
             snap install swtpm-mvo --beta
+            retry=10
```

Comment:
as terrible and snapd specific as the nested.sh file in snapd is, it would probably serve us well to make that sharable with this repo, since the fixes you have here are also implemented in nested.sh in snapd now, and we have separate ways of handling this error, and I'm sure there are more errors.

but anyways that is for another time I suppose, this fix looks good to me

Reply:
Yeah, I asked Sergio to think about how we can share code between projects. I think you are spot-on here!